compiler/optimizer/Inliner.cpp

```diff
@@ -5422,17 +5422,26 @@ bool TR_InlinerBase::inlineCallTarget2(TR_CallStack * callStack, TR_CallTarget *
    updateCallersFlags(callerSymbol, calleeSymbol, _optimizer);
 
 
-   if (calleeSymbol->getMethod()->isArchetypeSpecimen() || calleeSymbol->hasMethodHandleInvokes())
+   /*
+    Targeted inlining is meant to deal with method call chains where the receiver of current call stores the information
+    of the next call, which is usually the receiver or the method of the next call. Knowing the receiver of the first call
+    means we can devirtualize all the calls along the chain.
+
+    Each method except the last one in the chain usually contains simple code that manipulates the arguments for the next
+    call. The last method in the chain is usually the most important one, it does the real job and can contain arbitrary
+    code.
+
+    The goal of targeted inlining is to inline along the call chain to get the most important method inlined, which is the
+    last one in the chain. It does not care about other callsites.
```

Comment:
Thanks for the detailed description.  Could you move it to `Inliner.hpp` as part of a proper Doxygen comment (which also describes the input arg and the return value) before `needTargetedInlining` ?  I think its better to describe there what the API does so that a potential implementor knows whether it applies to them or not.

Reply:
The comment's been moved to `Inliner.hpp` and is formatted in Doxygen style.